rethinkdb/_export.py

```diff
@@ -226,6 +247,43 @@ def json_writer(filename, fields, task_queue, error_queue, format):
             pass
 
 
+def jsongz_writer(filename, fields, task_queue, error_queue, format, compression_level):
+    try:
+        with open(filename, "wb") as out:
+            # wbits 31 = MAX_WBITS + gzip header and trailer
+            compressor = zlib.compressobj(compression_level, zlib.DEFLATED, 31)
+            def compress_and_write(str):
+                out.write(compressor.compress(str.encode("utf-8")))
+
+            first = True
+            compress_and_write("[")
+            item = task_queue.get()
+            while not isinstance(item, StopIteration):
+                row = item[0]
+                if fields is not None:
+                    for item in list(row.keys()):
+                        if item not in fields:
+                            del row[item]
+                if first:
```

Comment:
This implies that the objects in the `JSON` array are each on a separate line.

I'm no compression expert, but since it'll be binary and unreadable from a high level perspective, why not skip the new line? Object would be written as follows
`[{...},{...},{...}...{...}]`
which would save n + 1 + 1 new lines (n-1 between each object, plus 2 for the first and the last, + 1 for the last row which `EOF` and is good. On huge table this could imply saving a considerable amount of bits. @gabor-boros do you have any knowledge about the topic?

Or maybe I'm wrong and the \n are used for compression. Let me know, I'm curious now.

Reply:
The code here is replicating what the json export does. `\n` are not used for compression. 

One can still unpack the jsongz file get the json file from inside (it's a standard gzip file), in which case the formatting might help. I considered the gains from removing the `\n`-s  marginal, but you might be right. If you have really small documents, the extra end lines might make a difference.

Reply:
Should say that the `import` has a custom json paarser (which I found weird, not sure what the reason for using a custom parser was, performance perhaps?) which might be affected by the lack of new lines (I expect it to be happy though).

Reply:
`if you have really small documents, the extra end lines might make a difference`

Did you mean big documents? Hehe

Regarding the custom json parser, I have no clue why there's a custom one. Probably, when the library has been written, there were parsers that did not fit/match the requirements. Nowadays there are tons of high performance parsers. In order to not break anything, I would keep the custom one for now.